Tests/BootCamp.Chapter.Tests/DeckTest.cs

```diff
@@ -0,0 +1,187 @@
+﻿using BootCamp.Chapter.Gambling;
+using FluentAssertions;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using Xunit;
+
+namespace BootCamp.Chapter.Tests
+{
+    public class DeckTest
+    {
+        [Fact]
+        public void Deck_Created_WithoutCards_Throws_ArgumentNullException()
+        {
+            List<Card> cards = null;
+
+            Action action = () => new Gambling.Deck(cards);
+
+            action.Should().Throw<ArgumentNullException>();
+        }
+
+        [Fact]
+        public void Draw_Given_DeckDeckWithNoCards_Throws_OutOfCardsException()
+        {
+            IDeck deck = new Gambling.Deck(new List<Card>());
+
+            Action action = () => deck.DrawFromTop();
+
+            action.Should().Throw<OutOfCardsException>();
+        }
+
+        [Theory]
+        [InlineData(-1)]
+        [InlineData(100)]
+        public void DrawAt_OutOfIndex_Throws_IndexOutOfRangeException(int index)
+        {
+            //Arrange
+            IDeck deck = BuildFullDeckOfCards();
+
+            //Act
+            Action action = () => deck.DrawAt(index);
+
+            //Assert
+            action.Should().Throw<IndexOutOfRangeException>();
+        }
+
+        [Fact]
+        public void DrawRandom_52Times_Returns_SameCards_DifferentOrder()
+        {
+            BuildListOfCardsGiveItToDeckAndCreateEmptyListOfCards(out List<Card>  cards, out IDeck deck, out List<Card> newCards);
+
+            for (int i = 0; i < 52; i++)
+            {
+                newCards.Add(deck.DrawRandom());
+            }
+
+            newCards.Should().Contain(cards).And.NotEqual(cards);
+        }
+
+        [Fact]
+        public void DrawRandom_Given_DeckWith1Card_returns_ExcpectedCard()
+        {
+
+            //Arrange
+            BuildDeckWith1Card(out Card lastCard, out IDeck deck);
+
+            //Act
+            Card cardDrawn = deck.DrawRandom();
+
+            //Assert
+            cardDrawn.Should().Be(lastCard);
+        }
+
+        [Fact]
+        public void DrawRandom_Given_DeckWithNoCards_Throws_OutOfCardsException()
```

Comment:
`When` not `Given`

Reply:
I disagree. When for input, given for preconditions.